daffodil-runtime1/src/main/scala/org/apache/daffodil/dpath/DFDLFunctions.scala

```diff
@@ -39,6 +41,31 @@ case class DFDLCheckConstraints(recipe: CompiledDPath) extends RecipeOpWithSubRe
   }
 }
 
+case class DFDLCheckRangeInclusive(recipe: CompiledDPath, rangeFrom: Int, rangeTo: Int)
+  extends RecipeOpWithSubRecipes(recipe) {
+
+  override def run(dstate: DState): Unit = {
+    recipe.run(dstate)
+    if (dstate.currentElement.valid.isDefined) {
+      dstate.setCurrentValue(dstate.currentElement.valid.get)
+    } else {
```

Comment:
You shouldn't need to check or set the value of ``valid`` here. That variable just says whether the the value of ``currentElement`` is valid with respect to the restrictions defined in the schema, which is unrelated to checking the if the value of node is between some range.

Ultimately, this function probably wants to look something like the below so that we run a recipe for both the data value and min/max value. Note that we need to save the currentNode when running multiple expressions:
```scala
val saved = dstate.currentNode
dataRecipe.run(dstate)
val dataVal = dstate.currentValue
dstate.setCurrentNode(saved)
minRecipe.run(dstate)
val minVal = dstate.currentValue
dstate.setCurrentNode(saved)
maxRecipe.run(dstate)
val maxVal = dstate.currentValue
val res = executeCheck(dataVal, minVal, maxVal)
dstate.setCurrentValue(res)
```